Update RFC with outcomes from blueprint discussion (#2302)
rerun-io/rerun: design/blueprint_store.md

    expand_panels=False,
    time_control=rrb.TimeControl(
        timeline="sim_time",
        play_state="paused",
    ),
    viewport=rrb.VerticalLayout(
        content=[
            rrb.View3D(
                root="world",
                content=[
                    rrb.Data("./points", visible_history=10).override(radius=0.1, color="blue"),
                    rrb.Data("./camera").override(image_plane=3),
                ],
            ),
            rrb.HorizontalLayout(
                content=[
                    rrb.View2D(
                        root="world/camera/image",
                        content=rrb.DataGroup(
                            rrb.Data("."),
                            rrb.Points2D("world/points"),
                        ).default(radius=0.2),
                    ),
                    rrb.ViewTimeSeries("metrics", content=rrb.RecursiveData(".")),
                ],
            ),
        ],
    ),
)

rr.spawn(blueprint)
```

The assorted objects used in blueprint construction are:
 - `App`: Container for top-level application state such as panel-visibility, menus, etc.
 - `TimeControl`: Specific state relevant to the time controls.

>  - `Tile`: Common base-class between `rrb.Container` and `rrb.SpaceView` (follows egui-tile-tree)

Reviewer: ```suggestion
 - `View`: Common base-class between `rrb.Container` and `rrb.SpaceView` (follows egui-tile-tree)
```
According to our concept map, this would be called "View". Containers are really "Container Views", although I think it's fine to call the python object `rrb.Container`

>  - `Tile`: Common base-class between `rrb.Container` and `rrb.SpaceView` (follows egui-tile-tree)

Reviewer: ```suggestion
 - `View`: Common base-class between `rrb.Container` and `rrb.SpaceView` (follows egui-tile-tree)
```
According to our concept map, this would be called "View". Containers are really "Container Views", although I think it's fine to call the python object `rrb.Container`

>     - `Container`: A layout of components (interface only)

Reviewer: ```suggestion
    - `Container`: A view that specifies layout of sub-views (interface only)
```

        - `HorizontalLayout`
        - `VerticalLayout`
        - ... additional layouts

>     - `SpaceView`: An actual view of data

Reviewer: ```suggestion
    - `SpaceView`: An actual view of data in a coordinate space
```

Reviewer 2: is a table view of raw data a coordinate space? I think that stretches it a bit too far

Reviewer: It does stretch it a bit far but I also think that is what the "Space" in SpaceView refers to. So that might be an argument to not call it space view.

Author: Yeah, coordinate-space isn't quite right, but there's some generalized analogy to coordinate-space that probably is, and in the mean-time it's a reasonable place-holder.  I think a Space is more loosely "a set of data that can all be transformed/mapped into a single *self-consistent* representation.

        - `View2D`
        - `View3D`
        - `ViewTimeSeries`
        - ... additional space-views

>  - `DataGroup`: A group of data entities with potentially shared overrides or defaults.
>     - `RecursiveData`: A special DataGroup that recursively includes all data entities under a given path.
>  - `Data`: A singular data entity.
>     - `Points2D`: A typed data entity for 2d points
>     - `Points3D`: A typed data entity for 3d points
>     - `Image`: A typed data entity for images
>     - ... additional typed data entities

Reviewer: ```suggestion
 - `Data`: A query that builds archetypes to draw in the space view
    - `Auto`: A query to automatically build archetypes from an entity path
    - `Points2D`: A query to build a Points2D archetype
    - `Points3D`: A query to build a Points3D archetype
    - `Image`: A query to build an Image archetype
    - ... additional typed archetype queries
 - `DataGroup`: A group of archetype queries with potentially shared overrides or defaults.
    - `RecursiveAuto`: A special DataGroup that recursively includes `Auto` queries for all entities under a given path.
```

Many Blueprint objects will allow for flexible upcasting into an obvious parent-type to reduce unnecessary typing. In
particular:
 - `Data` -> `DataGroup`
 - `[Data]` -> `DataGroup`
 - `DataGroup` -> `SpaceView` (If view category can be inferred)

>  - `Tile` -> `Viewport`

Reviewer: ```suggestion
 - `View` -> `Viewport`
```

 - `Viewport` -> `App`

This means a trivial expression like: `rr.show(rrb.Points3D("points"))` is still a valid Blueprint.

## Blueprint-Static Data

As a further simplification, the Blueprint will allow for the direct inclusion of static data, allowing users to bypass
the data-logging APIs entirely for simple use-cases that don't require temporal information.  This will be accomplished
by allowing `rrb.Data` objects to be constructured from any Rerun-loggable object.

> Data that is a *query* from the data store references an entity path used seprately by the logging APIs:

Reviewer 3: I guess we should call it the "recording store" to be consistent with the decision in todays standup (there are Recording Stores and Blueprint Stores)

```suggestion
Data that is a *query* from the recording store references an entity path used seprately by the logging APIs:
```

```python
# Log Data
for t in range(100):
    rr.set_time('step', t)
    rr.log("world/points", rr.Points3D(points))
...
# Construct Blueprint

> rrb.Data("/world/points")

Reviewer: ```suggestion
rrb.Auto("/world/points")
```
